gammapy/irf/io.py

```diff
@@ -124,3 +129,49 @@ def load_cta_irfs(filename):
     psf = EnergyDependentMultiGaussPSF.read(filename, hdu="POINT SPREAD FUNCTION")
 
     return dict(aeff=aeff, bkg=bkg, edisp=edisp, psf=psf)
+
+
+def load_irf_dict_from_file(filename):
+    """Open a fits file and generate a dictionary containing the Gammapy objects
+    corresponding ot the IRF components stored
+    
+    Parameters
+    ----------
+    filename : str, Path
+        path to the file containing the IRF components, if EVENTS and GTI HDUs 
+        are included in the file, they are ignored
+
+    Returns
+    -------
+    irf_dict : dict
+        dictionary with instances of the Gammapy obejcts corresponding 
+        to the IRF components        
+    """
+    filename = make_path(filename)
+
+    hdulist = fits.open(make_path(filename))
+    
+    irf_dict = {}
+
+    for hdu in hdulist:
+        hdu_class = hdu.header.get("HDUCLAS1", "").lower()
+        
+        if hdu_class == "response":
+            hdu_class = hdu.header.get("HDUCLAS4", "").lower()
+        
+            loc = HDULocation(
+                hdu_class=hdu_class,
+                hdu_name=hdu.name,
+                file_dir=filename.parent,
+                file_name=filename.name
+            )
+            
+            for name in HDUIndexTable.VALID_HDU_TYPE:
+                if name in hdu_class:
```

Comment:
Are you using `in` here to compare strings? Why not `==`? Also see the above comment.